client/my-sites/customer-home/controller.jsx

```diff
@@ -10,12 +10,19 @@ import page from 'page';
  */
 import CustomerHome from './main';
 import { getSelectedSiteSlug, getSelectedSiteId } from 'state/ui/selectors';
-import { canCurrentUserUseCustomerHome } from 'state/sites/selectors';
+import { canCurrentUserUseCustomerHome, getSitePlan } from 'state/sites/selectors';
 import isRecentlyMigratedSite from 'state/selectors/is-site-recently-migrated';
+import reducer from 'state/concierge/reducer';
+
+export default async function( context, next ) {
+	const state = await context.store.getState();
+	const siteId = await getSelectedSiteId( state );
+	const plan = await getSitePlan( state, siteId );
+
+	if ( ! plan.is_free && [ 'Business', 'eCommerce' ].includes( plan.product_name_short ) ) {
```

Comment:
🤔 I wonder if thisoptimization is worth it or not, we could also gate based on if the card is returned by the layout API. 

For simplicity sake, I'd probably be fine with dropping the check here and adding to the main reducer if Team Calypso doesn't object.

Reply:
Ah, OK, I can do that next week if we don't hear from anyone.

It _would_ fix another small issue I just found with this "check plans" method – but I'm uncertain it's an issue or not. The server-side logic for the card is "show if the current site has a Business or eCommerce plan, _or_ if the current user already has an appointment booked for _any_ site". BY moving the reducer to the root, we don't have to worry about the second case (which isn't covered in the current implementation).

But is that what we want? should Home show a user's appointment for _any_ site, or just the site being displayed? I could see user confusion both ways: "I thought I booked one of those appointments!" or "I don't have a plan on this site?..".

Reply:
We're going to only show the "existing session" version if it's for the current site.  I'll update that server-side, and add the reducer to the root reducer (as we haven't heard anything to the contrary).

Reply:
cb43e05